### DISCUSSION TITLE
Get ticks from host

### DISCUSSION THREAD

**Discussion by broadskins** [plugdata-team/plugdata#644] [Q&A]
Hey everyone, I've already been experimenting a bit, but I'm hoping someone can save me a lot more trial and error.

TL;DR - how can I get host tick values and bangs (960 per beat) while the host transport is running? I'm using Plugdata as a plugin in Logic.

I'm coming across from using Loomer's Architect, which I love but is very buggy. With this, the event loop runs every tick, 960 ticks per quarter, so 1,920 per second at 120bpm. This works well for the kind of things I'm interested in doing: complex rhythms and "real" sounding unquantized MIDI.

So far I've tried using the Playhead output, but this only updates per buffer, so with a buffer of 512 samples this only outputs 86 times a second, c.5% of the accuracy of ticks. I also tried using synced clocks set to output 960 times per beat of the main clock, but I'm guessing this isn't really what this is designed for, and I'm not sure whether it would be efficient or accurate.

Do tell me if this is not the kind of thing Plugdata is designed for, and if my paradigm is all wrong! I'm primarily interested in exploring using MIDI plugins to store and mutate MIDI loops within a DAW setting, with material either entered from DAW tracks or live.

**Comment by mugwortmugwort** [comment]
I'm interested in this also, I'm thinking to try set up a metro with the same interval as the projects bpm then offset it by some initial amount depending on the first playhead value received once play is pressed, so it doesn't need to keep reading from the DAWs position unless playback is start/stopped, may need some tweaking regarding latency issues but i would assume its more robust than trying to continually sync

**Comment by dromer** [comment]
If the plugin is tracking the host samplerate and blocksize, you should be able to get a bang with every cycle using `bang~` object.
